### PR TITLE
CORE-882: Add System.accountIsInitialized(...:completion)

### PR DESCRIPTION
**Pull request by EBGToo**
Asynchronously check if an account on a network is initialized.  Unlike accountIsInitialized(...), this functions does not cache the initialized status; it checks each time.  Works for *all* existing networks (BTC, ...).

### CODE REVIEW

File: WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java [hunk 1]
```diff
@@ -2402,6 +2402,45 @@ public boolean accountIsInitialized(com.breadwallet.crypto.Account account, com.
         return account.isInitialized(network);
     }
 
+    @Override
+    public void accountIsInitializedConfirm(com.breadwallet.crypto.Account account,
+                                            com.breadwallet.crypto.Network network,
+                                            CompletionHandler<Boolean, AccountInitializationError> handler) {
+        EXECUTOR_CLIENT.execute(() -> {
```
Comment (michael-brd) [inline WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java:2409]: Don't think this needs to be on an executor.

File: WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java [hunk 2]
```diff
@@ -2402,6 +2402,45 @@ public boolean accountIsInitialized(com.breadwallet.crypto.Account account, com.
         return account.isInitialized(network);
     }
 
+    @Override
+    public void accountIsInitializedConfirm(com.breadwallet.crypto.Account account,
```
Comment (michael-brd) [inline WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java:2406]: Is there a Swift equivalent? I'm confused.

Comment (EBGToo) [reply WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java:2406]: The Swift equivalent is `accountIsInitilialzed(_:onNetwork:completion)` where the third parameter distinguishes the function from `accountIsInitialized(_:onNetwork)`.

Comment (michael-brd) [reply WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java:2406]: Why not do the same here then? Why not overload `accountIsInitialized`?

File: WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java (hunk 1 above)
Comment (EBGToo) [reply WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java:2409]: Does this invocation of `handler` need to be in the CLIENT thread (one example from the body of the `network.getType()` case:
```
                   if (!publicKey.isPresent()) {
                        accountInitializeReportError(new AccountInitializationQueryError(new QueryNoDataError()), handler);
                        return;
                    }
```

File: WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java (hunk 2 above)
Comment (EBGToo) [reply WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java:2406]: Fixed.

File: WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java (hunk 1 above)
Comment (michael-brd) [reply WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java:2409]: Ah, I see now. So maybe the `EXECUTOR_CLIENT.execute(() -> {}` wrapping belongs in `accountInitializeReportSuccess/accountInitializeReportError`.

My original comment stemmed from the Swift/Java approach.

Comment (EBGToo) [reply WalletKitJava/corecrypto/src/main/java/com/breadwallet/corecrypto/System.java]: Updated.  Better.

**Review by carlcherry — APPROVED** [review]

**Comment by EBGToo** [comment]
[Gone a year unneeded; closing.]

**Pull request closed without merging**